tutorcairn/templates/cairn/build/cairn-clickhouse/Dockerfile

```diff
@@ -6,7 +6,20 @@ FROM docker.io/yandex/clickhouse-server:22.1.3.7
 # curl https://repo.yandex.ru/clickhouse/deb/stable/
 # The above command should be a 200, and not a 404.
 RUN rm /etc/apt/sources.list.d/clickhouse.list
-RUN apt update && apt install -y python3
+RUN apt update && apt install -y git build-essential libssl-dev zlib1g-dev \
+libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev \
+libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev
+
 COPY ./scripts /scripts
 RUN chmod a+x /scripts/*
-ENV PATH /scripts:${PATH}
+# Install pyenv
+# https://www.python.org/downloads/
+# https://github.com/pyenv/pyenv/releases
+ARG PYTHON_VERSION=3.12.2
+ENV PYENV_ROOT /opt/pyenv
+RUN git clone https://github.com/pyenv/pyenv $PYENV_ROOT --branch v2.3.36 --depth 1
+# Install Python
+RUN $PYENV_ROOT/bin/pyenv install $PYTHON_VERSION
+RUN $PYENV_ROOT/bin/pyenv global $PYTHON_VERSION
+
+ENV PATH /opt/pyenv/shims:/opt/pyenv/bin:/scripts:${PATH}
```

Comment:
why do we need 2 levels of path here?

Reply:
Without this, we are getting python **3.8** version when we check python version inside container.  Including both **/opt/pyenv/shims** and **/opt/pyenv/bin** ensures that both the shims and the actual Python binaries managed by pyenv are accessible and utilized correctly within the Docker container.

Reply:
We are pulling python 3.8 because the clickhouse-server image is based on ubuntu 22.04:

```
$ docker run --rm -it docker.io/clickhouse/clickhouse-server:24.1.8 uname -a
ClickHouse Database directory appears to contain a database; Skipping initialization
Linux 327ee7036a61 6.5.0-26-generic #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Mar 12 10:22:43 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```

Let's not attempt to install python 3.12 in this image. We should just wait until the base image is upgraded to ubuntu 24.04 upstream.

Moreover, note that the image was moved from docker.io/yandex/clickhouse-server to docker.io/clickhouse/clickhouse-server: https://hub.docker.com/r/clickhouse/clickhouse-server/tags

See this issue: https://github.com/overhangio/tutor-cairn/issues/30